.changeset/olive-sheep-beam.md

```diff
@@ -0,0 +1,6 @@
+---
+'@directus/api': patch
+---
+
+changed loop behavior from prematurely exiting to evaluating all subscriptions on ws; fixed issue where Directus might
+skip valid subscriptions if a filter didn't match records.
```

Comment:
```suggestion
Fixed an issue where multiple WebSocket subscriptions on same target might result in skipped events
```


In the api/src/websocket/handlers/subscribe.ts file, the behavior of the loop handling event dispatching to subscriptions was modified. Previously, if a subscription's filter didn't match any records, Directus would exit the entire loop, potentially skipping valid subsequent subscriptions. This behavior was rectified by changing the premature return statement to a continue statement, ensuring that Directus evaluates all relevant subscriptions regardless of the order they were registered in.